doc/classes/CodeEdit.xml

```diff
@@ -590,6 +590,12 @@
 		<member name="text_direction" type="int" setter="set_text_direction" getter="get_text_direction" overrides="TextEdit" enum="Control.TextDirection" default="1" />
 	</members>
 	<signals>
+		<signal name="new_line_added">
+			<param index="0" name="is_inserted_above" type="bool" />
+			<description>
+				Emitted when a new line has been inserted.
```

Comment:
This may seem obvious but descriptions are localized, too. Plus it's nice to spell it out.
```suggestion
				Emitted when a new line has been inserted. If [param is_inserted_above] is [code]true[/code], the new line has been added above the caret.
```
However, spelling it out highlights issues. Why is this signal exposed? It doesn't seem particularly useful as is because it's not possible to know where the new line has been added.

There could be another way to implement this PR without exposing a new signal (e.g through a Callable).
Alternatively, like properties and methods, I recall that prefixing a signal name with an underscore keeps it hidden from the API (thus making its usage not supported), but doing this is a last resort.

Reply:
I didn't add the line position as it can have multiple carets; that is why I get the lines directly inside the method I created while looping over all carets.

To be honest, I followed recommendation about using signal. But I can give it a try using a callback.

> Alternatively, like properties and methods, I recall that prefixing a signal name with an underscore keeps it hidden from the API (thus making its usage not supported), but doing this is a last resort.

Do you have an example in the code base I can have a look?

Thanks for the feedback!

Reply:
I don't see how a callback would fix the issues you mention, you could always add more arguments to the signal. From what I've seen, APIs use callback mostly in servers code, or when you need a return value (which is not the case here).

IMO, as long as the documentation describes how to get a list of caret positions, the signal could be useful when exposed as-is.

Reply:
I updated the signal documentation. Is it acceptable like this?